### PR TITLE
More stack options - on_failure / disable_rollback

### PR DESCRIPTION
**Pull request by herebebogans**
PR for Issue #20

* Add support for on_failure / disable_rollback at template_config and stack_config level.
* These options are invalid for update_stack api so are stripped in Bora::CFN::Stack.update (not sure if this is the best way).
* Added spec.

As the 2 options are mutually exclusive it feels better to leave AWS API to validate them?

### CODE REVIEW

File: spec/apply_spec.rb [hunk 1]
```diff
@@ -54,6 +54,25 @@
...
+
+        bora_config.templates.web.stacks.prod.merge!(stack_config)
+        expect(stack).to receive(:update)
+          .with(
+            hash_including(
+              :template_body,
+              'capabilities' => ['CAPABILITY_IAM']
+            )
+          )
+          .and_return(true)
```
Comment (ampedandwired) [inline spec/apply_spec.rb]: There's couple of problems here. The first is that this spec is probably not testing what you think it is... which is my fault for making things confusing as hell :)  Let me try to explain.

So there's two `stack.rb` files - `bora/stack.rb` and `bora/cfn/stack.rb` - I know, confusing and I should rename the latter to `cfn_stack.rb`. The change you made in this PR was in `bora/cfn/stack.rb`.

Now this particular spec is testing `bora/stack.rb`. The way it does that is by mocking out `bora/cfn/stack.rb` and asserting that it gets called the right way. So in effect the "stack" variable here is an instance of `Bora::Cfn::Stack`. The reason I did it this way is that mocking out the AWS API is a pain, so I decided to confine all that in `bora/cfn/stack.rb` and test it in isolation, and then for everything else just test that it calls it the right way.

So long story short, this test needs to be in `spec/cfn/stack_spec.rb` (around line 147). If you're not comfortable with that I'll happily implement the test myself.

The other problem (if we assume the first problem didn't exist) is that this test should probably explicitly check that certain hash keys were _not_ present using `hash_not_including`.

Comment (herebebogans) [inline spec/apply_spec.rb]: Thanks - will update this week.

**Comment by herebebogans** [comment]
Think I may need some help here :P .. struggling with the right approach to describe the stack double.

**Comment by ampedandwired** [comment]
Sorry for the delay, I'll have a bit of a look at this soon.

**Comment by ampedandwired** [comment]
This was really close, I just merged it and updated the spec myself - hope you don't mind. Sorry I haven't had much time to look at this lately.  Thanks again.
